protocol/x/clob/keeper/msg_server_cancel_orders.go

```diff
@@ -22,6 +26,20 @@ func (k msgServer) CancelOrder(
 
 	defer func() {
 		if err != nil {
+			// Gracefully handle the case where the order was already removed from state.
+			if errors.Is(err, types.ErrStatefulOrderDoesNotExist) {
+				processProposerMatchesEvents := k.Keeper.GetProcessProposerMatchesEvents(ctx)
+				removedOrderIds := lib.SliceToSet(processProposerMatchesEvents.RemovedStatefulOrderIds)
+				if _, found := removedOrderIds[msg.GetOrderId()]; found {
+					k.Keeper.Logger(ctx).Info(
+						errorsmod.Wrap(
+							err,
```

Comment:
Can we return a more informative error up from the handler (ideally one with an error code defined in https://github.com/dydxprotocol/v4-chain/blob/main/protocol/x/clob/types/errors.go)? Specifically telling the user the order existed but was fully-filled / removed in the last block.

Note that any potential deployments of this software would likely use this error in a block explorer. Ideally users can view the block explorer and the error is informative enough that they understand why their order was not actually canceled and potentially was partially filled.